sbin/spark-daemon.sh

```diff
@@ -124,9 +124,8 @@ if [ "$SPARK_NICENESS" = "" ]; then
 fi
 
 execute_command() {
-  local command="$@"
   if [ -z ${SPARK_NO_DAEMONIZE+set} ]; then
-      nohup -- $command >> $log 2>&1 < /dev/null &
+      nohup -- "$@" >> $log 2>&1 < /dev/null &
```

Comment:
It will lost single quotes around if use `local`. You can verify as follows:

```
set -x

execute_command() {
  echo "with local:"
  local command="$@"
  echo "no local:"
  "$@"
}

run_command() {
  execute_command ../bin/spark-submit "$@"
}
#!/usr/bin/env bash

run_command "$@"
```
Save above code as `SPARK-18645.sh` and run `sh SPARK-18645.sh --conf 'spark.driver.extraJavaOptions=-XX:+UseG1GC -XX:-HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/tmp'`
you will see like this:
```
+ run_command --conf 'spark.driver.extraJavaOptions=-XX:+UseG1GC -XX:-HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/tmp'
+ execute_command ../bin/spark-submit --conf 'spark.driver.extraJavaOptions=-XX:+UseG1GC -XX:-HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/tmp'
+ echo 'with local:'
with local:
+ local 'command=../bin/spark-submit --conf spark.driver.extraJavaOptions=-XX:+UseG1GC -XX:-HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/tmp'
+ echo 'no local:'
no local:
+ ../bin/spark-submit --conf 'spark.driver.extraJavaOptions=-XX:+UseG1GC -XX:-HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/tmp'
...
```